docs/proposals/non-default-namespace-deployment.md

```diff
@@ -0,0 +1,145 @@
+# Proposal: Supporting non default namespace CSI deployment at Openshift 
+
+Last updated: 09.12.2021
+
+## Abstract
+
+This proposal contains approaches for supporting CSI deployment in non default namespace at Openshift platform, 
+introducing Privileged SCC at CSI ServiceAccounts.`
+
+## Background
+
+In case of CSI deployment at Openshift platform in non default namespace, CSI pods have Restricted SCC by default.   
+But restricted denies access to all host features and resources like disk. Must be Privileged SCC because it allows
+access to all privileged and host features and resources.
+For give privileged SCC to custom user, the following command need to be executed:
+`oc adm policy add-scc-to-user privileged -z <user name> -n <namespace>`
+And so, we should give this scc for CSI's service accounts, that need it: 
+ * csi-node-sa;
+ * csi-baremetal-extender-sa.
+
+## Proposal
+
+In basic there are two approaches that allows us to support this feature:
+1. Move logic of creation csi's sa to csi-baremetal-operator and in case of deployment to Openshift platform - give the
```

Comment:
How will that work for user who doesn't have admin privileges?

Reply:
The question is about which user? If it's about operator's service account, then it need only the following role to assign scc:
https://github.com/dell/csi-baremetal/pull/645/files#diff-6d7d994f70a983b2f669ca3449d6d261007f2483daf9312a0190a8f36bf55ce0R37